edi_storage_oca/components/base.py

```diff
@@ -40,15 +43,15 @@ def _dir_by_state(self, direction, state):
             (self.backend[direction + "_dir_" + state] or "").strip().rstrip("/")
         )
 
-    def _remote_file_path(self, direction, state, filename):
-        """Return remote file path by direction and state for give filename.
-
-        :param direction: string stating direction of the exchange
-        :param state: string stating state of the exchange
-        :param filename: string for file name
-        :return: PurePath object
-        """
-        return self._dir_by_state(direction, state) / filename.strip("/ ")
+    def _get_remote_file_path(self, state, filename=None):
+        """Retrieve remote path for current exchange record."""
+        filename = filename or self.exchange_record.exchange_filename
+        direction = self.exchange_record.direction
+        directory = self.backend[direction + "_dir_" + state] or ""
```

Comment:
pls keep using `self._dir_by_state` here which is what you are doing here but w/out validation

Reply:
we should probably keep the old method and log deprecation in case someone is still using it